test/model/message_list_test.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/api/route/messages.dart';
+import 'package:zulip/model/message_list.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+import '../api/fake_api.dart';
+import '../model/binding.dart';
+import '../model/test_store.dart';
+import '../example_data.dart' as eg;
+
+const int userId = 1;
+const int streamId = 2;
+
+Future<PerAccountStore> setupStore(ZulipStream stream) async {
+  await TestZulipBinding.instance.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+  PerAccountStore store = await TestZulipBinding.instance.globalStore.perAccount(eg.selfAccount.id);
+  store.addUser(eg.user(userId: userId));
+  store.addStream(stream);
+  return store;
+}
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+  const narrow = StreamNarrow(streamId);
+
+
+  group('update message tests', () {
```

Comment:
When I run the whole test file, I get the following failure message:
```
$ flutter test test/model/message_list_test.dart 
00:01 +1 -1: update message tests update events are correctly applied to message [E]
  'package:zulip/model/store.dart': Failed assertion: line 122 pos 12: '!_accounts.containsKey(account.id)': is not true.
  dart:core                              _AssertionError._throwNew
  package:zulip/model/store.dart 122:12  GlobalStore.insertAccount
  

To run this test again: /home/greg/n/flutter/flutter/bin/cache/dart-sdk/bin/dart test /home/greg/z/flutterz/test/model/message_list_test.dart -p vm --plain-name 'update message tests update events are correctly applied to message'
00:01 +1 -1: Some tests failed.                                                     
```

Not sure why the stack trace is truncated there. But if you navigate to the call sites of that `GlobalStore.insertAccount` method, there's one over in the login flow which shouldn't be relevant and there's one in `TestGlobalStore.add`; and the latter is called in your `setupStore` function:
```dart
  await TestZulipBinding.instance.globalStore.add(eg.selfAccount, eg.initialSnapshot());
```
So that's presumably the stack.

Since you're getting the failure only when running both tests together, and not when running either individually, it's presumably because some state is leaking from the first test and needs to be cleaned up.

Reply:
Aha, bingo, that's it - I was setting up/inserting the same account twice. Will put a fix up in a sec.